src/Microsoft.ML.AutoML/TrainerExtensions/TrainerExtensionCatalog.cs

```diff
@@ -58,13 +59,24 @@ public static ITrainerExtension GetTrainerExtension(TrainerName trainerName)
         }
 
         public static IEnumerable<ITrainerExtension> GetTrainers(TaskKind task,
-            IEnumerable<TrainerName> whitelist)
+            IEnumerable<TrainerName> whitelist, ColumnInformation columnInfo)
         {
             IEnumerable<ITrainerExtension> trainers;
             if (task == TaskKind.BinaryClassification)
             {
                 trainers = GetBinaryLearners();
             }
+            else if (task == TaskKind.MulticlassClassification &&
+                columnInfo.ImagePathColumnNames.Count == 1 &&
+                columnInfo.CategoricalColumnNames.Count == 0 &&
+                columnInfo.NumericColumnNames.Count == 0 &&
+                columnInfo.TextColumnNames.Count == 0)
+            {
+                // Image Classification case where all you have is a label column and image column.
+                // This trainer takes features column of vector of bytes which will not work with any
+                // other trainer.
+                return new List<ITrainerExtension>() { new ImageClassificationExtension() };
+            }
             else if (task == TaskKind.MulticlassClassification)
```

Comment:
The fall through is normal multi-class. We should add image handling there. 

I'd recommend two options for the fall-thru case: (for when there are images+text/numerics/categoricals)
1. Returning the penultimate layer from the `ImageClassification` trainer (running as a featurizer)
2. Running as the 1st model in a stack ensemble (returning the `score` column as features)

Other tasks, like regression/clustering, etc still need image support. Option 1, above, won't cover it as `ImageClassification` requires a classification label. Perhaps a label-less, no fine-tuning, featurizer mode? Basically using your `ImageClassification` as a replacement for DNNImageFeaturizer.

Reply:
We are not handling Image handling here because ImageClassificationTrainer does not support just returning featurized outputs and your approach actually causes a crash in the code, as there is a bug that fails to load the saved model @bigmiao.zhang@gmail.com can comment more. It is not within the scope of this PR to add your approach or handle cases other than where you just have images and labels, the scope is to integrate ImageClassificationTrainer API into AutoML.

---
In reply to: [339329937](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339329937) [](ancestors = 339329937)